src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/test/IntegrationTests/CodeGenerationIntegrationTest.cs

```diff
@@ -508,6 +508,49 @@ public class AllTagHelper : {typeof(TagHelper).FullName}
             AssertLinePragmas(compiled.CodeDocument, designTime: false);
         }
 
+        [Fact]
+        public void ViewComponentTagHelperOptionalParam_Runtime()
+        {
+            // Arrange
+            AddCSharpSyntaxTree($@"
+using System;
+
+public class OptionalTestViewComponent
+{{
+    public string Invoke(bool showSecret = false)
+    {{
+        return showSecret ? ""what a secret"" : ""not a secret"";
+    }}
+}}
+public class OptionalTestWithParamViewComponent
+{{
+    public string Invoke(string secret, bool showSecret = false)
+    {{
+        var isSecret = showSecret ? ""what a secret"" : ""not a secret"";
+        return isSecret + "" : "" + secret;
+    }}
+}}
+public class OptionalWithMultipleTypesViewComponent
+{{
+    public string Invoke(int age = 42, double favoriteDecimal = 12.3, char favoriteLetter = 'b', DateTime? birthDate = null)
```

Comment:
Could you also add a reference type? e.g.

`
```suggestion
    public string Invoke(int age = 42, double favoriteDecimal = 12.3, char favoriteLetter = 'b', DateTime? birthDate = null, string street = null)
```